front/pages/api/poke/workspaces/[wId]/invitations.ts

```diff
@@ -20,6 +28,52 @@ type PokePostInvitationResponseBody = {
   error_message?: string;
 };
 
+async function getExistingMember(workspace: WorkspaceType, email: string) {
+  const userModel = await User.findOne({ where: { email } });
+  if (userModel) {
+    const user = renderUserType(userModel);
+    const membership =
+      await MembershipResource.getLatestMembershipOfUserInWorkspace({
+        user,
+        workspace,
+      });
+    if (membership) {
+      return user;
+    }
+  }
+
+  return null;
+}
+
+async function handleMembershipUpdate(
+  workspace: WorkspaceType,
+  user: UserType,
+  role: ActiveRoleType
+) {
+  const r = await MembershipResource.updateMembershipRole({
+    user,
+    workspace,
+    newRole: role,
+    // We allow to re-activate a terminated membership when updating the role here.
+    allowTerminated: true,
```

Comment:
If I understand correctly, this will re-use the latest active membership, I think in this case even though it comes from Poke, we want to create a new membership to have some transparency in the admin UI.

Reply:
I took the same logic as in admin, which uses this : https://github.com/dust-tt/dust/blob/bf747292138f7d621aa0ea4f3d00b055bcb055cb/front/pages/api/w/%5BwId%5D/members/%5BuId%5D/index.ts#L116 . It creates a new membership, even if the current one is terminated.